src/assertion.c

```diff
@@ -0,0 +1,329 @@
+    }
+    if(cb0r_vlen(auth_data) > sizeof(ca->auth_data)) {
+        return FIDO_ERR_BUFFER_TOO_SHORT;
+    }
+    memcpy(ca->auth_data, cb0r_value(auth_data), cb0r_vlen(auth_data));
+    return FIDO_OK;
+}
+
+static int cbor_assert_decode_signature(const cb0r_t signature, fido_assert_reply_t *ca) {
+    if (!cbor_bytestring_is_definite(signature)) {
+        return FIDO_ERR_CBOR_UNEXPECTED_TYPE;
+    }
+
+    if(cb0r_vlen(signature) > sizeof(ca->signature)) {
+        return FIDO_ERR_BUFFER_TOO_SHORT;
+    }
+    memcpy(ca->signature, cb0r_value(signature), cb0r_vlen(signature));
+    return FIDO_OK;
+}
+
+static int cbor_assert_decode_large_blob_key(const cb0r_t large_blob_key, fido_assert_reply_t *ca) {
+    if (!cbor_bytestring_is_definite(large_blob_key)) {
+        return FIDO_ERR_CBOR_UNEXPECTED_TYPE;
+    }
+
+    if(cb0r_vlen(large_blob_key) > sizeof(ca->large_blob_key)) {
+        return FIDO_ERR_BUFFER_TOO_SHORT;
+    }
+    memcpy(ca->large_blob_key, cb0r_value(large_blob_key), cb0r_vlen(large_blob_key));
+    ca->has_large_blob_key = true;
+    return FIDO_OK;
+}
+
+/**
+ * @brief Parse an entry of the authenticatorGetInfo CBOR map.
+ *
+ * @param key The cb0r element representing the map key
+ * @param value The cb0r element representing the map value
+ * @param arg User-passed argument (here: CBOR info).
+ * @return int FIDO_OK if entry could be parsed.
+ */
+static int parse_get_assert_reply_entry(const cb0r_t key, const cb0r_t value, void *arg) {
+    if (key->type != CB0R_INT || key->value > UINT8_MAX) {
+        // Just ignore the entry according to https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-20210615.html#message-encoding.
+        return FIDO_OK;
+    }
+
+    fido_assert_reply_t *ca = (fido_assert_reply_t*)arg;
+
+    switch (key->value) {
+        case 1: // credential
+            return cbor_iter_map(value, cbor_assert_decode_credential, ca);
+        case 2: // auth data
+            return cbor_assert_decode_auth_data(value, ca);
+        case 3: // signature
+            return cbor_assert_decode_signature(value, ca);
+        case 4: // user --ignore for now
+            return FIDO_OK;
+        case 5: // numberOfCredentials --ignore for now
+            return FIDO_OK;
+        case 6: // userSelected --ignore for now
+            return FIDO_OK;
+        case 7: // large blob key
+            return cbor_assert_decode_large_blob_key(value, ca);
+        default: // ignore
+            fido_log_debug("%s: cbor type", __func__);
+            return FIDO_OK;
+    }
+}
+
+// TODO: function to set rpid
+// TODO: function to set client data (and compute hash)
+
+static int fido_dev_get_assert_tx(
+    fido_dev_t *dev,
+    fido_assert_t *assert,
+    const es256_pk_t *pk,
+    const fido_blob_t *ecdh
+) {
+    // 32 > 1 byte command + 1 byte map header + 1 byte get key + max. 9 byte get value + 1 byte offset key + max. 9 byte offset value
```

Comment:
Should be adjusted to the new calculation.

Reply:
Automatically calculates the buffer size now. Not especially a speed increase but not the end of the world, either.
At least, it is future-proof for extensions.